tests/test_bst.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from structures.bst import BST, BSTNode
+import pytest
+
+
+@pytest.fixture
+def create_bst():
+    btree = BST()
+
+    n1 = BSTNode(val=1)
+    n2 = BSTNode(val=2)
+    n3 = BSTNode(val=3)
+    n4 = BSTNode(val=4)
+    n5 = BSTNode(val=5)
+    n6 = BSTNode(val=6)
+
+    btree.root = n4
+    btree._size = 6
+
+    n4.left_child = n2
+    n4.right_child = n6
+
+    n2.parent = n4
+    n6.parent = n4
+
+    n2.left_child = n1
+    n2.right_child = n3
+
+    n1.parent = n2
+    n3.parent = n2
+
+    n6.left_child = n5
+
+    n5.parent = n6
+
+    return btree
+
+
+# will insert the value val into the BST.  If val is already
+# present, it will be ignored.
+def test_insert():
+    # A new tree should have nothing in its root.
+    btree = BST()
+    assert btree.root is None
+
+    # First insertion should set root to that node, and that node
+    # should have no parents or children.
+    btree.insert(val=7)
+    assert btree.root is not None
+    assert btree.root.val == 7
+    assert btree.root.parent is None
+    assert btree.root.left_child is None
+    assert btree.root.right_child is None
+
+    # When inserting a duplicate value, nothing should appear to happen.
+    # No errors, no creating new nodes.
+    btree.insert(val=7)
+    assert btree.root.left_child is None
+    assert btree.root.right_child is None
+
+    # When inserting a value smaller than the root, it should go in a new
+    # node as left_child to root.
+    btree.insert(val=4)
+    assert btree.root.left_child is not None
+    assert btree.root.right_child is None
+    assert btree.root.left_child.val == 4
+
+    # When inserting a value larger than the root, it should go in a new
+    # node as right_child to root.
+    btree.insert(val=10)
+    assert btree.root.right_child is not None
+    assert btree.root.right_child.val == 10
+
+    # This is smaller than root's right_child, and should go in that node's
+    # left_child slot.
+    btree.insert(val=8)
+    assert btree.root.right_child.left_child is not None
+    assert btree.root.right_child.left_child.val == 8
```

Comment:
This test is too large.  Break it into smaller tests, each of which tests one of the comment-separated chunks of functionality.  (insert, insert duplicate, insert smaller, insert larger)